driver_darwin.go

```diff
@@ -42,6 +43,8 @@ type audioInfo struct {
 }
 
 type driver struct {
+	ctx           context.Context
+	cancel        context.CancelFunc
```

Comment:
Instead of `context.Context`, using a channel with an empty struct should be fine. `d.cancel` will be just closing the channel.

Reply:
channel here may not as stable and easy to use here. especially when we are in a asynchronous environment.

here is another solution using  channel

https://github.com/rcrai/oto/commit/abd2da457df4e57887625cb3ad9e4fe4ac1d88a6

And it will crashed like this:

```
$ go run main.go 
panic: close of closed channel

goroutine 6 [running]:
github.com/hajimehoshi/oto.(*driver).Close(0xc000116000, 0x0, 0x0)
        /Workspace/src/github.com/hajimehoshi/oto/driver_darwin.go:253 +0xa5
github.com/hajimehoshi/oto.(*driverWriter).Close(0xc00007e1b0, 0x0, 0x0)
        /Workspace/src/github.com/hajimehoshi/oto/context.go:170 +0xb5
github.com/hajimehoshi/oto.(*Context).Close(0xc00000e048, 0xc00007e1b0, 0x4112528)
        /Workspace/src/github.com/hajimehoshi/oto/context.go:106 +0x7b
github.com/hajimehoshi/oto.NewContext.func1(0xc00000e048)
        /Workspace/src/github.com/hajimehoshi/oto/context.go:89 +0xa5
created by github.com/hajimehoshi/oto.NewContext
        /Workspace/src/github.com/hajimehoshi/oto/context.go:84 +0x269
exit status 2
```


Yes, I may add a few flags to avoid this issue, but I guess it is not grace enough.

Reply:
What about setting the channel nil after closing?

But anyway I understanding using context is easier. Let's go with it.